src/N3DataFactory.js

```diff
@@ -200,6 +202,12 @@ export function termFromId(id, factory) {
     return factory.literal(id.substr(1, endPos - 1),
             id[endPos + 1] === '@' ? id.substr(endPos + 2)
                                    : factory.namedNode(id.substr(endPos + 3)));
+  case '<':
+    // Parse quad
+    let result = id.match(/^<<("(?:""|[^"])*"[^ ]*|[^ ]+) ("(?:""|[^"])*"[^ ]*|[^ ]+) ("(?:""|[^"])*"[^ ]*|[^ ]+) ?("(?:""|[^"])*"[^ ]*|[^ ]+)?>>$/);
+    result = result.splice(1).filter((x) => x !== undefined).map((id) => termFromId(unEscape(id), factory));
```

Comment:
The `splice`/`filter`/`map` combination creates three array loops; better to have a `for` (starting from 1)

Reply:
The undefined check is better done as `typeof x !== 'undefined'`. (Compilers are optimized for this.)